pkg/stores/proxy/proxy_store.go

```diff
@@ -422,20 +427,30 @@ func (s *Store) Create(apiOp *types.APIRequest, schema *types.APISchema, params
 	}
 
 	name := types.Name(input)
-	ns := types.Namespace(input)
-	if name == "" && input.String("metadata", "generateName") == "" {
-		input.SetNested(schema.ID[0:1]+"-", "metadata", "generatedName")
+	namespace := types.Namespace(input)
+	generateName := input.String("metadata", "generateName")
+
+	if name == "" && generateName == "" {
+		input.SetNested(schema.ID[0:1]+"-", "metadata", "generateName")
 	}
-	if ns == "" && apiOp.Namespace != "" {
-		ns = apiOp.Namespace
-		input.SetNested(ns, "metadata", "namespace")
+
+	if attributes.Namespaced(schema) && namespace == "" {
+		if apiOp.Namespace == "" {
+			return nil, nil, validation.ErrorCode{
+				Status: http.StatusUnprocessableEntity,
+				Code:   errNamespaceRequired,
+			}
```

Comment:
Sorry I didn't pick this up on my previous review:

```json
{
  "type": "error",
  "links": {},
  "code": "metadata.namespace is required",
  "message": "",
  "status": 422
}
```

It think to be more in line with the current errors we return it should look like:

```json
{
  "type": "error",
  "links": {},
  "code": "InvalidBodyContent",
  "message": "metadata.namespace is required",
  "status": 422
}
```

For example, we do the following in other places in Steve: https://github.com/rancher/steve/blob/bdb38275a6852620aa0f8de2f83420274dfd1f31/pkg/schema/definitions/handler.go#L23-L30.

So it would look something like this:

```go
return nil, nil apierror.NewAPIError(invalidBodyContent, "metadata.namespace is required")
```

where invalidBodyContent is already defined here: https://github.com/rancher/wrangler/blob/f9725ad6a04e725835441bc897b7b637f4c7e5ee/pkg/schemas/validation/error.go#L29. I'm not sure if we need a custom ErrorCode or can re-use the InvalidBodyContent so I'd suggest having a look at current usage of InvalidBodyContent if that makes sense here.

Reply:
@tomleb Just updated. Another look ?